File: backend/app/models/analysis.py
"""
拆解报告数据模型
"""
from sqlalchemy import Column, String, JSON, DateTime
from app.models.base import BaseModel


class AnalysisReport(BaseModel):
    """拆解报告表"""
    __tablename__ = "analysis_reports"
    
    video_url = Column(String(500), nullable=False, unique=True, index=True)
    video_info = Column(JSON, nullable=True)
    basic_info = Column(JSON, nullable=True)
    shot_table = Column(JSON, nullable=True)
    golden_3s = Column(JSON, nullable=True)
    highlights = Column(JSON, nullable=True)
    viral_formula = Column(JSON, nullable=True)
    keywords = Column(JSON, nullable=True)
    production_tips = Column(JSON, nullable=True)

File: frontend/src/api/analysis.ts
import apiClient from './client'

export interface AnalysisReport {
  id: string
  video_url: string
  video_info?: any
  basic_info?: any
  shot_table?: any[]
  golden_3s?: any
  highlights?: any[]
  viral_formula?: any
  keywords?: any
  production_tips?: any
  techniques?: any[]
  created_at: string
}

export interface AnalysisListResponse {
  total: number
  items: AnalysisReport[]
  limit: number
  offset: number
}

export const analysisApi = {
  // 获取报告列表
  getReports(params?: {
    video_url?: string
    limit?: number
    offset?: number
  }) {
    return apiClient.get<AnalysisListResponse>('/analysis/reports', { params })
  },

  // 获取报告详情
  getReportDetail(id: string) {
    return apiClient.get<AnalysisReport>(`/analysis/reports/${id}`)
  },

  // 分析视频
  analyzeVideo(video_url: string, options?: any) {
    return apiClient.post('/analysis/analyze', {
      video_url,
      options
    })
  },

  // 批量分析
  batchAnalyze(video_urls: string[]) {
    return apiClient.post('/analysis/batch', video_urls)
  }
}

